ouroboros-consensus-shelley-test/src/Test/Consensus/Shelley/Generators.hs

```diff
@@ -505,3 +505,14 @@ instance Arbitrary (STS.PredicateFailure (STS.LEDGER TPraosMockCrypto)) where
 instance Arbitrary (STS.PredicateFailure (STS.LEDGERS TPraosMockCrypto)) where
   arbitrary = genericArbitraryU
   shrink    = genericShrink
+
+{-------------------------------------------------------------------------------
+  Versioned generators for serialisation
+-------------------------------------------------------------------------------}
+
+-- We only have single version, so no special casing required.
+--
+-- This blanket orphan instance will have to be replaced with more specific
+-- ones, once we introduce a different Shelley version.
+instance (Arbitrary v, Arbitrary a) => Arbitrary (WithVersion v a) where
```

Comment:
Replace with specific instances because we'll import this module in the generators for Cardano, causing overlaps with the Byron ones.

Better solution: add a block argument to `WithVersion` so we can still write a blanket instance for Shelley.